### PR TITLE
Added ProcessLauncherWhitelist

### PR DESCRIPTION
**Pull request by mikehall-ms**

### CODE REVIEW

**Comment by msalehmsft** [comment]
+1

**Comment by ooeygui** [comment]
Do we need to have a localize version?

File: ProcessLauncherWhitelist/ProcessLauncherWhitelist/ProcessLauncherWhitelist.cpp [hunk 1]
```diff
@@ -0,0 +1,706 @@
...
+	DWORD dwSize = 0;
+	if (ERROR_SUCCESS != RegQueryValueEx(hKey, wcValue, NULL, NULL, NULL,&dwSize))
+	{
+		RegCloseKey(hKey);
+		return;
+	}
+
+	// allocate the buffer
+	wchar_t *lpValues = (wchar_t*)malloc(dwSize);
+	ZeroMemory(lpValues, dwSize);
```
Comment (ooeygui) [inline ProcessLauncherWhitelist/ProcessLauncherWhitelist/ProcessLauncherWhitelist.cpp]: While this is likely safe, do you want to move it after the null check?

File: ProcessLauncherWhitelist/ProcessLauncherWhitelist/ProcessLauncherWhitelist.cpp [hunk 2]
```diff
@@ -0,0 +1,706 @@
...
+void WriteRegFile()
+{
+	if (Whitelist.size() == 0)
+	{
+		wprintf(L"No whitelist items to write to the registry file\n");
+		return;
+	}
+
+	// start with writing "IoTWhitelist.reg", we can prompt later...
+	HANDLE hFile = CreateFile(L"IoTWhiteList.reg", GENERIC_WRITE, 0, NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
```
Comment (ooeygui) [inline ProcessLauncherWhitelist/ProcessLauncherWhitelist/ProcessLauncherWhitelist.cpp]: Check for INVALID_HANDLE_VALUE?

**Comment by ooeygui** [comment]
:+1:

**Pull request merged**